knowledge_repo/repositories/gitrepository.py

```diff
@@ -503,15 +503,28 @@ def __remote_host(self):
         return None
 
     @property
+    def __remote_port(self):
+        port = 22
+        if self.git_has_remote:
+            m = re.match(r'^(.*?)?@([^/:]*):?(\d+)?', self.git_remote.url)
+            if m:
+                if m.group(3):
+                    port = m.group(3)
+        try:
```

Comment:
You shouldn't need this try block, since the regex filter already restricts it to digits. Given that `\d` actually matches non western digit glyphs as well, it might be worth change it to: `[0-9]+` instead of `\d+`.